modules/serverless-ossm-installing-and-configuring-openshift-service-mesh-3-x.adoc

```diff
@@ -0,0 +1,236 @@
+= Installing and configuring {SMProductShortName} 3.x
+
+[role="_abstract"]
+You can integrate {SMProductShortName} 3.x with {ServerlessProductShortName} by installing and configuring the required Istio components, gateways, and Knative Serving resources. Once these resources are configured, you can deploy the Knative Serving instance with Istio enabled to ensure that your serverless workloads run within the {SMProductShortName} environment. 
+
+.Procedure
+
+. Create an `Istio` resource in the `istio-system` namespace with the following configuration:
++
+[source,yaml]
+----
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  values:
+    meshConfig:
+      defaultConfig:
+        terminationDrainDuration: 35s
+  updateStrategy:
+    inactiveRevisionDeletionGracePeriodSeconds: 30
+    type: InPlace
+  namespace: istio-system
+  version: v1.26-latest
+----
+
+. Apply the resource by running the following commands:
++
+[source,terminal]
+----
+$ oc new-project istio-system
+----
++
+[source,terminal]
+----
+$ oc apply -f istio.yaml
+----
+
+. Create an `IstioCNI` resource in the `istio-cni` namespace with the following configuration:
++
+[source,yaml]
+----
+apiVersion: sailoperator.io/v1
+kind: IstioCNI
+metadata:
+  name: default
+spec:
+  namespace: istio-cni
+  version: v1.26-latest
+----
+
+. Apply the resource by running the following commands:
++
+[source,terminal]
+----
+$ oc new-project istio-cni
+----
++
+[source,terminal]
+----
+$ oc apply -f istio-cni.yaml
+----
+
+. Create a file named `gateway-deploy.yaml` with the following configuration:
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: knative-istio-ingressgateway
+  namespace: knative-serving-ingress
+spec:
+  selector:
+    matchLabels:
+      knative: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway <1>
+      labels:
+        knative: ingressgateway <2>
+        sidecar.istio.io/inject: "true" <3>
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto <4>
+
+---
+# Set up roles to allow reading credentials for TLS
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: istio-ingressgateway-sds
+  namespace: knative-serving-ingress
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: istio-ingressgateway-sds
+  namespace: knative-serving-ingress
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: istio-ingressgateway-sds
+subjects:
+  - kind: ServiceAccount
+    name: default
+----
++
+<1> Select the gateway injection template rather than the default sidecar template.
+<2> Set a unique label for the gateway. This is required to ensure Gateways can select this workload.
+<3> Enable gateway injection.
+<4> The image will automatically update each time the pod starts.
```

Comment:
```suggestion
* `spec.template.metadata.annotations.inject.istio.io/templates` specifies the gateway injection template rather than the default sidecar template.
* `spec.template.metadata.labels.knative` defines a unique label for the gateway. This is required to ensure Gateways can select this workload.
* `spec.template.metadata.labels.sidecar.istio.io/inject` enables gateway injection.
* `spec.template.spec.containers.image` ensures that the image automatically updates each time the pod starts.
```
Refining this to follow the new style of [writing commands in code blocks](https://redhat-documentation.github.io/supplementary-style-guide/#explain-commands-variables-in-code-blocks).